rest/OrderService.js

```diff
@@ -56,12 +60,27 @@ class OrderService {
             throw new IllegalArgumentError(EMPTY_ARGUMENT_ERR);
         }
     }
+    
+    #checkProductIsActive(productId) {
+        const product =  ProductPersistence.getById(productId);
+        if (!product || !product.isActive) {
```

Comment:
Why do you check if the product is defined when right before calling this method in create you call assertProductExists anyway?